### ISSUE TITLE
Question about xgb_v1.py

### ISSUE THREAD

**Issue by Superhzf** [sadz2201/Kaggle_Allstate#1]
Congrats and thanks for sharing! I am learning about your idea. 

I have a question about your code but I don't know where to ask.

`num_1s=full.ix[:,0:51].sum(axis=1)`

Why do you just calculate the number of 1 from column 0 to column 50?

**Comment by sadz2201** [comment]
Hi Zach, 

I'm glad you asked. There's actually an interesting story how I came to choose only column 0 to 50 for that calculation.

Initially I had `num_1s=full.ix[:,0:72].sum(axis=1)`, as there are 72 logical columns. 
Later I had tried some dimension reduction stuff & eliminated 21 columns that were less important, & modified the above line to `full.ix[:,0:51]`. By doing so the CV score was a lot worse, so I had to undo my changes & ditch the dimension reduction stuff. Somehow I forgot to change it back to `full.ix[:,0:72]` , but surprisingly the CV score as well as LB score was better :-). 

Basically I made a small blunder, but it resulted in slightly better score so I decided to stick with it. LOL :-)
Hope this answers your question.

**Comment by Superhzf** [comment]
Thanks for answering. I understand this point now.

I noticed that you did PCA to reduce dimensions from 72 logical columns and kept the first five principal components, then added these principal components as new columns. 

Why didn't you delete those 72 logical columns, did it make the result worse?

**Comment by sadz2201** [comment]
That's right. Dropping the 72 columns & using only the 5 PCA components made the results worse. But adding the PCA components as additional features along with the 72 columns gave some improvement.
